### PR TITLE
Dynamically show Location facet when a whitelist of building facet …

### LINKED ISSUE #1174: Show Location facet when Art Library selected
- [ ] Create a whitelist of libraries that will show the Location facet. "Art & Architecture (Bowes)" is currently the only library, but others may follow.
- [ ] When the whitelisted library is selected in the Building facet, display the Location facet:
  - at the top of the facets list
  - open (same behaviour as Databases facet)

### PR DESCRIPTION
**Pull request by jkeck**
…values are selected.

Closes #1174 

art-locked-stacks-facet

### CODE REVIEW

**Comment by jkeck** [comment]
@lmcglohon or @ndushay can you comment here to confirm once the config change is in production?

Thanks!

**Comment by ndushay** [comment]
@jkeck   just put the config changes on SW solr master;  the changes will propagate to slaves with the next index update (within the hour, I expect).

**Comment by ndushay** [comment]
@jkeck @lmcglohon   config changes are on all 3 SW solr slaves.  Green light.

**Comment by jkeck** [comment]
:+1: thanks! This is ready for review then.

File: app/controllers/concerns/location_facet.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  def building_facet_includes_library_with_sublocation?
+    params[:f] &&
+      params[:f][:building_facet] &&
+      params[:f][:building_facet].any? do |facet|
+        libraries_with_sublocations.include?(facet)
+      end
+  end
+
+  def libraries_with_sublocations
```
Comment (ndushay) [inline app/controllers/concerns/location_facet.rb]: another constant disguised as a method

**Comment by ndushay** [comment]
What do you think of grabbing the Solr schema from https://github.com/sul-dlss/solrmarc-sw/blob/master/stanford-sw/solr/conf/schema.xml -- either as a one off copy now, or make the Rake task use it from there?  solrconfig-no-repl.xml   would be the solr config you'd want.

File: app/controllers/concerns/location_facet.rb (hunk 1 above)
Comment (jkeck) [inline app/controllers/concerns/location_facet.rb]: :+1:

**Comment by jkeck** [comment]
@ndushay I think it would be ideal to automate that in some way (and I'm not sure I have the bandwidth right now to take that on).

I'm not sure having the separate config is causing us any harm at this point, so I'm inclined to keep it as is.  If  you would like, please feel free to create a ticket for us to evaluate doing that in a future sprint.

**Comment by ndushay** [comment]
Created issue  #1212.   No problem postponing it.

**Pull request merged**